packages/taco/src/conditions/context/context.ts

```diff
@@ -213,37 +149,30 @@ export class ConditionContext {
   }
 
   public async toJson(): Promise<string> {
-    const parameters = await this.toObj();
+    const parameters = await this.toContextParameters();
     return toJSON(parameters);
   }
 
-  public withCustomParams(
-    params: Record<string, CustomContextParam>,
-  ): ConditionContext {
-    return new ConditionContext(
-      this.provider,
-      this.condition,
-      params,
-      this.signer,
-    );
-  }
-
-  public async toWASMContext(): Promise<Context> {
+  public async toCoreContext(): Promise<CoreContext> {
     const asJson = await this.toJson();
-    return new Context(asJson);
+    return new CoreContext(asJson);
   }
 
+  public toContextParameters = async (): Promise<Record<string, ContextParam>> => {
```

Comment:
> More of a general question and something I may have missed - how is the above done by apps?
@derekpierre They would have to call this context method. We can make it more user-friendly.

Reply:
This function is only available *after* `ConditionContext` creation and it returns the populated values. 

I'm wondering how I, as the app developer, know what context variables are needed for the condition in the ciphertext I'm about to decrypt - so that I know what context variable values I need to provide to the `decrypt(...)` call. (implicit v explicit).

`ConditionContext` has the `requiredParmeters` property on an instance, but to create an instance of `ConditionContext`, a customParameters dictionary must be provided (constructor, or fromConditions).

Reply:
Decryptor is given `ThresholdMessageKit` which is then made into `ConditionContext` internally. So, not sure what is the best way to expose this to the user-facing API: `ConditionContext.requiredParameters(messageKit)`?

I will look into it.